[3810] Preview Search Panel

aachavarria opened the pull request:
https://github.com/craftercms/craftercms/issues/3810

aachavarria committed:
content.ts: Fixing issue with GUEST_CHECK_IN;
ToolsPanel.tsx: Fixing issue when no site selected;
PreviewSearchPanel.tsx: Implementing orOperator;

aachavarria committed:
Adding LevelDescriptor.tsx icon;
ToolsPanel.tsx no site validation;

rart commented on ui/app/src/components/Icons/LevelDescriptor.tsx:
```diff
@@ -0,0 +1,32 @@
+      <clipPath id="clip0">
+        <rect width="12" height="12" fill="white" transform="translate(0.0344238 0.68457)" />
+      </clipPath>
+    </defs>
```
Can we get rid of everything but the paths?

aachavarria replied:
yes

rart commented on ui/app/src/components/PreviewSearchPanel/PreviewSearchPanel.tsx:
```diff
@@ -133,7 +130,8 @@ function SearchResults(props) {
+  orOperator: true
   // sortBy: '_score',
   // sortOrder: 'desc',
   // filters: {}
```
?

rart commented on ui/app/src/components/PreviewSearchPanel/PreviewSearchPanel.tsx:
```diff
@@ -144,69 +142,81 @@ const mimeTypes = ['image/png', 'image/jpeg', 'image/gif', 'video/mp4', 'image/s
-      //     setError(response);
-      //   }
-      // )
+      // pipe(takeUntil(unMount$),switchMap())
```
?

aachavarria committed:
Changes on PreviewSearchPanel.tsx

aachavarria committed:
BulkUpload Updates;
Adding LevelDescriptor.tsx icon;
Changes on Host to update editMode when DnD starts;
Adding minimized to ToolsPanelEmbeddedAppViewButton.tsx

rart commented on ui/app/src/components/Dialogs/BulkUploadDialog.tsx:
```diff
@@ -629,13 +629,6 @@ export default function BulkUploadDialog(props: BulkUploadProps) {
-      dispatch(emitSystemEvent(itemCreated({ target: path })));
-    },
-    [dispatch]
-  );
```
This isn't much help and takes away from the presentational nature of the dialog UI. Let's rollback.

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/3810

aachavarria committed:
ToolsPanelEmbeddedAppViewButton.tsx: Using DialogHeader instead of PanelToolHeader;
Guest.tsx: Changes on EditMode to add a new State instead of using Status;

rart commented on ui/app/src/components/SystemStatus/MinimizedBar.tsx:
```diff
@@ -40,8 +40,9 @@ interface MinimizedBarProps {
+
   return (
-    <Paper className={classes.root}>
+    <Paper className={classes.root} id="testeo">
```
x

aachavarria committed:
Rollback on BulkUploadDialog.tsx

aachavarria committed:
missing prop;

aachavarria committed:
Removing hardCoded ID

rart commented on ui/app/src/components/ToolsPanelEmbeddedAppViewButton/ToolsPanelEmbeddedAppViewButton.tsx:
```diff
@@ -56,14 +60,48 @@ const useStyles = makeStyles((theme) =>
 export default function ToolsPanelEmbeddedAppViewButton(props) {
   const [open, setOpen] = useState(false);
   const openEmbeddedApp = () => setOpen(true);
```
Please check what happens when you...
1. open dialog
2. minimize it
3. click again on the tool panel item to open

rart commented on ui/app/src/modules/Preview/Tools/ToolPanel.tsx:
```diff
@@ -63,6 +82,16 @@ export const PanelHeader: FunctionComponent<PanelHeaderProps> = (props) => {
+          <IconButton onClick={onClose}>
+            <CloseIcon />
+          </IconButton>
+        )}
```
This file was meant to be rolled back

aachavarria committed:
ToolPanel.tsx rollback;

aachavarria committed:
adding uiKey to widgetDescriptor;
using hook usePossibleTranslation to get Widget titles;

aachavarria committed:
Fix on DialogHeader.tsx;
Hidden Translation option on ItemMenu;

aachavarria committed:
typo

rart committed:
Update BulkUploadDialog.tsx

rart commented on ui/app/src/components/PreviewComponentsPanel/PreviewComponentsPanel.tsx:
```diff
@@ -119,7 +126,8 @@ export const ComponentsPanelUI: React.FC<ComponentsPanelUIProps> = (props) => {
             {
-              id: 'craftercms.components.PreviewBrowseComponentsPanel'
+              id: 'craftercms.components.PreviewBrowseComponentsPanel',
+              uiKey: `uiKey_${Date.now()}`
```
Add the uiKey on the reducer when not present

rart committed:
Update MinimizedBar.tsx

rart commented on ui/guest/src/store/reducers/root.ts:
```diff
@@ -732,7 +722,8 @@ const reducerFunctions: {
     hostCheckedIn: true,
-    status: action.payload.editMode ? EditingStatus.LISTENING : EditingStatus.OFF
+    editMode: action.payload.editMode,
+    status: EditingStatus.LISTENING
```
No need to change the status on HostCheckIn

rart commented on ui/app/src/services/configuration.ts:
```diff
@@ -196,19 +197,20 @@ export function setActiveTargetingModel(data): Observable<ActiveTargetingModel>
       if (xml) {
         const widgets = xml.querySelector('[id="craftercms.components.ToolsPanel"] > configuration > widgets');
         if (widgets) {
+          widgets.querySelectorAll('widget').forEach((e, index) => e.setAttribute('uiKey', String(index)));
```
Please add comment explaining why we had to add this:

When rendering widgets dynamically and changing pages on the tools panel, if there are duplicate react `key` props across pages, react may no swap the components correctly, incurring in unexpected behaviours. We need a unique key for each widget.

Add comment to the createwidgetdescriptor function too.

rart commented on ui/app/src/modules/Preview/Tools/ToolPanel.tsx:
```diff
@@ -28,6 +28,8 @@ type ToolPanelProps = PropsWithChildren<{
   BackIcon?: ElementType;
   onBack?: () => void;
+  onMinimize?: () => void;
+  onClose?: () => void;
```
x

rart commented on ui/app/src/modules/Preview/Tools/ToolPanel.tsx:
```diff
@@ -47,6 +49,9 @@ const useStyles = makeStyles((theme: Theme) =>
+    },
+    alignRight: {
+      marginLeft: 'auto'
     }
```
x

rart commented on ui/app/src/modules/Preview/Tools/ToolPanel.tsx:
```diff
@@ -73,6 +78,7 @@ export function ToolPanel(props: ToolPanelProps): ReactElement | null {
   const dispatch = useDispatch();
   const { formatMessage } = useIntl();
   const { title, BackIcon, onBack = () => dispatch(popToolsPanelPage()), classes } = props;
+
```
x

rart commented on ui/app/src/modules/Preview/Tools/ToolPanel.tsx:
```diff
@@ -73,6 +78,7 @@ export function ToolPanel(props: ToolPanelProps): ReactElement | null {
+
   return (
     <>
       <PanelHeader
```
This whole file should disappear from the PR files (check)

rart commented on ui/app/src/components/ToolsPanelPageButton/ToolsPanelPageButton.tsx:
```diff
@@ -25,6 +25,7 @@ export default function ToolsPanelPageButton(props) {
     dispatch(
       pushToolsPanelPage({
         id: 'craftercms.components.ToolsPanelPage',
+        uiKey: `uiKey_${Date.now()}`,
```
👆🏻 use create widget descriptor method

aachavarria committed:
creating createWidgetDescriptor to generate uuid

aachavarria committed:
Merge remote-tracking branch 'aaron/feature/3810' into feature/3810

aachavarria committed:
removing unnecessary change
